src/main/java/de/siphalor/amecs/api/KeyModifier.java

```diff
@@ -1,32 +1,56 @@
 package de.siphalor.amecs.api;
 
-import de.siphalor.amecs.impl.AmecsAPI;
 import org.apache.commons.lang3.ArrayUtils;
 
+import de.siphalor.amecs.impl.AmecsAPI;
+import de.siphalor.amecs.impl.ModifierPrefixTextProvider;
+import net.minecraft.client.util.InputUtil;
+
 @SuppressWarnings("WeakerAccess")
-public class KeyModifier {
-	public static final KeyModifier NONE = new KeyModifier("none", -1);
-	public static final KeyModifier ALT = new KeyModifier("alt", 0, 342, 346);
-	public static final KeyModifier CONTROL = new KeyModifier("control", 1, 341, 345);
-	public static final KeyModifier SHIFT = new KeyModifier("shift", 2, 340, 344);
+public enum KeyModifier {
```

Comment:
Okay. I understand the problem with enums. But I am using the static `VALUES` field. So IF some mod wants to add a custom modifier it can simply create a new instance of the enum (via reflection) and add it to the `VALUES` array. And then it will be used verywhere. (Other mods which are iterating these enums should of course also use the `VALUES` field)

Reply:
Well my original comment was just an explanation — it was not meant as criticism :)

If an enum is what we go for — then I'd just accept that it won't be extendable. No workarounds with enum byte code manipulation ^^.

Reply:
> then I'd just accept that it won't be extendable.

As I said. It is extendable. But it is a bit harder now. (It does not require bytecode manipulation. Just using java.reflect)